exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/validate/BatchValidator.java

```diff
@@ -293,14 +292,20 @@ private void validateVector(String name, ValueVector vector) {
       // structure to check.
     } else if (vector instanceof BaseRepeatedValueVector) {
       validateRepeatedVector(name, (BaseRepeatedValueVector) vector);
-    } else if (vector instanceof RepeatedMapVector) {
-      validateRepeatedMapVector(name, (RepeatedMapVector) vector);
+    } else if (vector instanceof AbstractRepeatedMapVector) {
```

Comment:
Could you please clarify your comment below, for example

>Checking ```AbstractRepeatedMapVector``` includes DictVector, in order to avoid NPE while getting value from ```NullReader```. The ```NullReader``` may be returned when index points to the absent value in dict array, example query ```SELECT dict_array[3].keyInDict ... ``` .

Reply:
I've updated the comment - hope it is clearer